node/service/src/chain_spec/moonriver.rs

```diff
@@ -0,0 +1,300 @@
+pub fn generate_accounts(mnemonic: String, num_accounts: u32) -> Vec<AccountId> {
+	let childs = derive_bip44_pairs_from_mnemonic::<ecdsa::Public>(&mnemonic, num_accounts);
+	debug!("Account Generation");
+	childs
+		.iter()
+		.map(|par| {
+			let account = get_account_id_from_pair::<ecdsa::Public>(par.clone());
+			debug!(
+				"private_key {} --------> Account {:x?}",
+				sp_core::hexdisplay::HexDisplay::from(&par.clone().seed()),
+				account
+			);
+			account
+		})
+		.flatten()
+		.collect()
+}
+
+/// Generate a chain spec for use with the development service.
+pub fn development_chain_spec(mnemonic: Option<String>, num_accounts: Option<u32>) -> ChainSpec {
+	// Default mnemonic if none was provided
+	let parent_mnemonic = mnemonic.unwrap_or(
+		"bottom drive obey lake curtain smoke basket hold race lonely fit walk".to_string(),
+	);
+	let mut accounts = generate_accounts(parent_mnemonic, num_accounts.unwrap_or(10));
+	// We add Gerald here
+	accounts.push(AccountId::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap());
+	ChainSpec::from_genesis(
+		"Moonriver Development Testnet",
+		"development",
+		ChainType::Development,
+		move || {
+			testnet_genesis(
+				AccountId::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap(),
+				// Validator
+				vec![(
+					AccountId::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap(),
+					None,
+					1_000 * GLMR,
+				)],
+				moonbeam_inflation_config(),
+				accounts.clone(),
+				Default::default(), // para_id
+			)
+		},
+		vec![],
+		None,
+		None,
+		Some(serde_json::from_str("{\"tokenDecimals\": 18}").expect("Provided valid json map")),
+		Extensions {
+			relay_chain: "dev-service".into(),
+			para_id: Default::default(),
+		},
+	)
+}
+
+/// Generate a default spec for the parachain service. Use this as a starting point when launching
+/// a custom chain.
+pub fn get_chain_spec(para_id: ParaId) -> ChainSpec {
+	ChainSpec::from_genesis(
+		// TODO Apps depends on this string to determine whether the chain is an ethereum compat
+		// or not. We should decide the proper strings, and update Apps accordingly.
+		// Or maybe Apps can be smart enough to say if the string contains "moonbeam" at all...
+		"Moonriver Development Testnet",
+		"local_testnet",
+		ChainType::Local,
+		move || {
+			testnet_genesis(
+				AccountId::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap(),
+				// Validator
+				vec![(
+					AccountId::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap(),
+					None,
+					1_000 * GLMR,
+				)],
+				moonbeam_inflation_config(),
+				vec![AccountId::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap()],
+				para_id,
+			)
+		},
+		vec![],
+		None,
+		None,
+		Some(serde_json::from_str("{\"tokenDecimals\": 18}").expect("Provided valid json map")),
+		Extensions {
+			relay_chain: "local_testnet".into(),
+			para_id: para_id.into(),
+		},
+	)
+}
+
+pub fn moonbeam_inflation_config() -> InflationInfo<Balance> {
```

Comment:
This is the same for all runtime chain_specs. Let's make another rust file for the shared logic between all chain_specs (if we follow the feature flag approach).

Reply:
chain_specs are still WIP